src/main/java/com/adwicorp/aanandamsn/traceconfig/TraceIdAspect.java

```
@@ -0,0 +1,37 @@
+package com.adwicorp.aanandamsn.traceconfig;
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.After;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.UUID;
+
+@Aspect
+@Component
+public class TraceIdAspect {
+
+    private static final Logger logger = LoggerFactory.getLogger(TraceIdAspect.class);
+
+    @Before("within(@org.springframework.web.bind.annotation.RestController *)")
+    public void captureTraceId(JoinPoint joinPoint) {
+        HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.getRequestAttributes()).getRequest();
+        String traceId = request.getHeader("X-Trace-ID");
+        if (traceId == null) {
```

if trace id is not provided in request then throw error that mandatory parameter is not provided.

---

Done